src/xmipp/libraries/reconstruction_starpu/mpi/mpi_reconstruct_fourier_starpu.cpp

```diff
@@ -0,0 +1,456 @@
+/***************************************************************************
+ *
+ * Authors:     Jan Polak (456647@mail.muni.cz)
+ *              David Strelak (davidstrelak@gmail.com)
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#include <cstdint>
+#include <vector>
+#include <deque>
+#include <semaphore.h>
+
+#define STARPU_DONT_INCLUDE_CUDA_HEADERS
+#include <starpu.h>
+
+#include "../reconstruct_fourier_util.h"
+#include "../reconstruct_fourier_starpu_util.h"
+
+#include "mpi_reconstruct_fourier_starpu.h"
+
+/** DEVELOPMENT: Enable to display logs from this file */
+const bool mpiLogging = false;
+
+#define LOG_MPI(COUT) do { if (mpiLogging) {\
```

Comment:
Probably because it used to contain some more preprocessor magic (`__LINE__`?) or maybe I just got used to LOG macros (which are macros for that preprocessor magic), I'm not sure. But you are right, it is no longer necessary for it to be a macro.

Reply:
Now I remember why it is a macro - `COUT` is not just a simple variable, but can be used to print multiple different things, i.e.: `LOG_MPI("batch query received from " << recvStatus.MPI_SOURCE << " (" << complete << " completed)");`. It will have to stay as a macro, because alternatives are slow/complicated (unless I'm missing something).